packages/docusaurus-plugin-content-blog/src/index.ts

```diff
@@ -145,15 +150,15 @@ export default async function pluginContentBlog(
       });
 
       const blogListPaginated: BlogPaginated[] = paginateBlogPosts({
-        blogPosts,
+        blogPosts: listedBlogPosts,
         blogTitle,
         blogDescription,
         postsPerPageOption,
         basePageUrl: baseBlogUrl,
       });
 
       const blogTags: BlogTags = getBlogTags({
-        blogPosts,
+        blogPosts: listedBlogPosts,
```

Comment:
This is not ideal because when an unlisted post gets browsed, it may display a tag for which the tag page does not actually exist in prod (ie click tag => 404)

An unlisted page should lead to the creation of the related tag page. 

If a tag only contains unlisted blog posts, then the tag should become itself unlisted and be filtered from the tag index page.